synapse/storage/data_stores/main/relations.py

```diff
@@ -31,7 +31,7 @@
 
 
 class RelationsWorkerStore(SQLBaseStore):
-    @cached(tree=True)
+    @cached(tree=True, max_entries="event_cache_size")
```

Comment:
So we're changing this to match the event cache size rather than allowing it to be configured separately? This feels like something that should be considered separately (and we should think how to support things that have already configured these).

Reply:
in the installs I've seen it, the cache for `get_relations_for_event` and such needs to be closer to the event cache size than zero, since a lot of times in sync or whatever, for each event it will also look for the relations and aggregations for it. It was the major thrashing candidate.

Reply:
I can believe that, though on matrix.org I've noticed that that's not true on a lot of workers. I'd be tempted to remove it and we can add it back with a way of overriding this behaviour if you set a per cache factor override.